staging/src/k8s.io/dynamic-resource-allocation/structured/allocator_test.go

```diff
@@ -1152,6 +1152,104 @@ func TestAllocator(t *testing.T) {
 				deviceAllocationResult(req0, driverA, pool1, device2, true),
 			)},
 		},
+		"all-devices-some-allocating-admin-access": {
```

Comment:
This test case is new to handle the `Device in use in same claim` case inside `allocateDevice`. I played with this for a bit but couldn't find a way to end up with a solvable case where that's hit, so the test case ends up with no expected allocations.